index.bs

```diff
@@ -364,11 +400,54 @@ When asked to <dfn export>report the layout shift</dfn> for an active
     1. Set |newEntry|'s <dfn attribute for=LayoutShift>hadRecentInput</dfn>
         attribute to <code>true</code> if {{LayoutShift/lastInputTime}} is less
         than 500 milliseconds in the past, and <code>false</code> otherwise.
+    1. Set |newEntry|'s <dfn attribute for=LayoutShift>sources</dfn> attribute
+        to the result of invoking the algorithm to <a>report the layout shift
+        sources</a> for |D|.
     1. <a href="https://w3c.github.io/performance-timeline/#dfn-queue-a-performanceentry">Queue the PerformanceEntry</a>
         |newEntry| object.
 
 </div>
 
+Report the layout shift sources {#sec-report-layout-shift-sources}
+------------------------------------------------------------------
+
+<div algorithm="report the layout shift sources">
+When asked to <dfn export>report the layout shift sources</dfn> for an active
+{{Document}} |D|, run the following steps:
+
+1. Let |S| be any set which satisfies these constraints:
```

Comment:
nit: set of Nodes? Also, I would prefer we provide a more concrete algorithm to obtain `S`, as this is a bit vague regarding how to construct the set.

Reply:
The fact that the set contains nodes is implied by the first constraint.  The specification here is deliberately agnostic to the method of implementation, and allows that more than one set is possible.  I think we should allow the user agent some degree of flexibility in the method of computation.  I also think the functional description is easier to understand than a procedural description.

Reply:
Agree that a more explicit algorithm would be better - it won't necessarily constrain user agents in how they calculate the set (as long as it's not observably different from the algorithm), but will provide a baseline algorithm that they can then improve.

Reply:
Thanks for the feedback.  I'll think about a better way to express this.